### PR TITLE
feat(dcs/redis): add a resource to manage DCS Redis instances within HuaweiCloud.

### LINKED ISSUE #1375: [DCS] Add separated Redis and Memcached instance support
I'd like to use specific Redis and Memcached instance resource to be more clear.

### PR DESCRIPTION
**Pull request by chengxiangdong**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1375

**Special notes for your reviewer**:

**Release note**:

```release-note
add huaweicloud_dcs_redis_instance to manage the DCS Redis instances, and support to change the instance capacity.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dcs' TESTARGS='-run TestAccResourceDcsRedisInstance'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dcs -v -run TestAccResourceDcsRedisInstance -timeout 360m -parallel 4
=== RUN   TestAccResourceDcsRedisInstance_basic
=== PAUSE TestAccResourceDcsRedisInstance_basic
=== RUN   TestAccResourceDcsRedisInstance_masterStandby
=== PAUSE TestAccResourceDcsRedisInstance_masterStandby
=== RUN   TestAccResourceDcsRedisInstance_cluster
=== PAUSE TestAccResourceDcsRedisInstance_cluster
=== RUN   TestAccResourceDcsRedisInstance_rwsplit
=== PAUSE TestAccResourceDcsRedisInstance_rwsplit
=== CONT  TestAccResourceDcsRedisInstance_basic
=== CONT  TestAccResourceDcsRedisInstance_rwsplit
=== CONT  TestAccResourceDcsRedisInstance_cluster
=== CONT  TestAccResourceDcsRedisInstance_masterStandby
--- PASS: TestAccResourceDcsRedisInstance_masterStandby (312.01s)
--- PASS: TestAccResourceDcsRedisInstance_rwsplit (323.20s)
--- PASS: TestAccResourceDcsRedisInstance_cluster (368.11s)
--- PASS: TestAccResourceDcsRedisInstance_basic (536.41s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dcs       536.640s
```

### CODE REVIEW

File: docs/resources/dcs_redis_instance.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+* `vpc_id` - (Required, String, ForceNew) The ID of VPC which the instance belongs to.
+  Changing this creates a new instance resource.
+
+* `subnet_id` - (Required, String, ForceNew) The ID of subnet which the instance belongs to.
+  Changing this creates a new instance resource.
+
+* `security_group_id` - (Optional, String) The ID of the security group which the instance belongs to.
+  This parameter is mandatory for Redis 3.0 version.
+
+* `ip` - (Optional, String, ForceNew) The IP address of the DCS instance, which can only be the currently available IP
```
Comment (Jason-Zhang9309) [inline docs/resources/dcs_redis_instance.md:121]: How about use `pravite_ip` or `fixed_ip` which is more clear?
And the resource may support public ip in the future, use ip here could be confusing.

File: docs/resources/dcs_redis_instance.md [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+  You can specify an available IP for the Redis instance (except for the Redis Cluster type).
+  If omitted, the system will automatically allocate an available IP address to the Redis instance.
+  Changing this creates a new instance resource.
+
+* `port` - (Optional, Int) Port customization, which is supported only by Redis 4.0 and Redis 5.0 instances.
+  Default value is 6379.
+
+* `password` - (Optional, String) Specifies the password of a DCS instance.
+  The password of a DCS Redis instance must meet the following complexity requirements:
+  + Enter a string of 8 to 32 bits in length.
```
Comment (Jason-Zhang9309) [inline docs/resources/dcs_redis_instance.md]: Must be a string of 8 to 32 bits in length.

File: docs/resources/dcs_redis_instance.md [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+  + The new password cannot be the same as the old password.
+  + Must contain three combinations of the following four characters: Lower case letters, uppercase letter, digital,
```
Comment (Jason-Zhang9309) [inline docs/resources/dcs_redis_instance.md:133]: Move this rule to Line 133 could be better

File: docs/resources/dcs_redis_instance.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+    For example, if the start time is 22:00:00, the end time is 02:00:00.
+  + Parameters `maintain_begin` and `maintain_end` must be set in pairs.
+  + If parameter maintain_end is left blank, parameter maintain_begin is also blank.
+    In this case, the system automatically allocates the default end time 06:00:00.
+
+* `backup_policy` - (Optional, List) Specifies the backup configuration to be used with the instance.
+  The structure is described below.
+
+* `rename_commands` - (Optional, Map, ForceNew) Critical command renaming, which is supported only by Redis 4.0 and
+  Redis 5.0 instances and not by Redis 3.0 instance.
```
Comment (Jason-Zhang9309) [inline docs/resources/dcs_redis_instance.md]: but not by

File: docs/resources/dcs_redis_instance.md [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - A resource ID in UUID format.
+
+* `status` - Cache instance status. The valid values are as follows:
+  + `CREATING`: Creating is the status before the Running state.
+  + `CREATEFAILED`: The DCS instance failed to be created.
```
Comment (Jason-Zhang9309) [inline docs/resources/dcs_redis_instance.md]: CREATING and CREATEFAILED should not be here

File: docs/resources/dcs_redis_instance.md (hunk 1 above)
Comment (chengxiangdong) [reply docs/resources/dcs_redis_instance.md:121]: it follows the old resource, it is named ip in the resource.

File: docs/resources/dcs_redis_instance.md (hunk 5 above)
Comment (chengxiangdong) [reply docs/resources/dcs_redis_instance.md]: done

File: docs/resources/dcs_redis_instance.md (hunk 4 above)
Comment (chengxiangdong) [reply docs/resources/dcs_redis_instance.md]: done

File: docs/resources/dcs_redis_instance.md (hunk 3 above)
Comment (chengxiangdong) [reply docs/resources/dcs_redis_instance.md:133]: done

File: docs/resources/dcs_redis_instance.md (hunk 2 above)
Comment (chengxiangdong) [reply docs/resources/dcs_redis_instance.md]: done

File: huaweicloud/services/acceptance/dcs/resource_huaweicloud_dcs_redis_instance_test.go [hunk 6]
```diff
@@ -0,0 +1,520 @@
...
+				Config: testAccResourceDcsRedisInstance_basic(
+					map[string]string{
+						"name":       name,
+						"version":    "5.0",
+						"capacity":   "8",
+						"specCode":   "redis.ha.xu1.large.p2.8",
+						"chargeMode": "prePaid",
+						"periodUnit": "month",
+						"autoRenew":  "false",
+						"period":     "1",
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/dcs/resource_huaweicloud_dcs_redis_instance_test.go]: chargeMode, periodUnit, autoRenew,period are not used in testAccResourceDcsRedisInstance_basic

Comment (chengxiangdong) [reply huaweicloud/services/acceptance/dcs/resource_huaweicloud_dcs_redis_instance_test.go]: done

**Pull request closed without merging**